site/content/en/latest/install/install-argocd.md

```diff
@@ -0,0 +1,139 @@
++++
+title = "Install with Argo CD"
+weight = -99
++++
+
+[Argo CD](https://argo-cd.readthedocs.io) is a declarative, GitOps continuous delivery tool for Kubernetes.
+Argo CD can be used to manage the deployment of Envoy Gateway on Kubernetes clusters.
+
+## Before you begin
+
+{{% alert title="Compatibility Matrix" color="warning" %}}
+Refer to the [Version Compatibility Matrix](/news/releases/matrix) to learn more.
+{{% /alert %}}
+
+{{< boilerplate kind-cluster >}}
+
+Argo CD must be installed in your Kubernetes cluster, and the argocd CLI must be available on your local machine.
+If you haven’t set it up yet, you can follow the [official installation guide](https://argo-cd.readthedocs.io/en/stable/operator-manual/installation/) to install Argo CD.
+
+## Install with Argo CD
+
+Create a new Argo CD Application that pulls the Envoy Gateway Helm chart as its source.
+
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: envoy-gateway
+  namespace: argocd
+spec:
+  project: default
+  source:
+    chart: gateway-helm
+    repoURL: docker.io/envoyproxy
+    targetRevision: {{< helm-version >}}
+    helm:
+      valuesObject:
+        certgen:
+          rbac:
+            annotations:
+              argocd.argoproj.io/sync-wave: "-1"  # Ensure rbac is created before the certgen job.
```

Comment:
https://github.com/envoyproxy/gateway/pull/5352 hit this issue and set `"helm.sh/hook-weight": "-1"`, can we add that first into cert gen and avoid this `patch` setting in argo install docs ?